### PR TITLE
Adding missing tests for some edge cases

### PR DESCRIPTION
**Pull request by jabrena**
Reviewing the behaviour of `Mono`, I noticed that for some edge cases with null values, the TestFile didn´t have some scenarios, so I added them.

Juan Antonio

### CODE REVIEW

File: reactor-core/src/test/java/reactor/core/publisher/MonoJustTest.java [hunk 1]
```diff
@@ -50,13 +50,26 @@ public void normal() {
 	                .verifyComplete();
     }
 
+	@Test
```
Comment (simonbasle) [inline reactor-core/src/test/java/reactor/core/publisher/MonoJustTest.java]: this is not needed. `Mono.just` emits whatever value is passed to it (`null` being an invalid value) without interpreting the value in any way. the optional could be a `String` or anything. The second test you added makes a little bit of sense though, although technically `Optional.empty()` (tested further down) and `Optional.ofNullable(null)` are the same.

Comment (jabrena) [reply reactor-core/src/test/java/reactor/core/publisher/MonoJustTest.java]: Hi @simonbasle, I have just removed the first test.

**Comment by simonbasle** [comment]
Thanks for the PR 👍

**Comment by jabrena** [comment]
Thanks you for this fantastic library.
I will try to help more in the future. I will invest time in the Reactor usage.

Cheers

Juan Antonio
